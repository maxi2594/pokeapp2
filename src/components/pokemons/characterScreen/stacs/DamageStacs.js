import React from 'react'
import PropTypes from 'prop-types';

import { ExelColums } from '../../../commons/ExelColums';

export const DamageStacs = ({ damage }) => {

    const damageArr = Object.entries(damage)

    return (
        <div className='typeStacs'>
            <div className='statsTitle'>
                <h4>Damage Stacs</h4>
            </div>
            {
                damageArr.map(elem =>
                (<ExelColums
                    key={elem[0]}
                    item={elem[0]}
                    values={elem[1]}
                />)
                )
            }

        </div>
    )
}
DamageStacs.propTypes = {
    damage: PropTypes.object.isRequired
}