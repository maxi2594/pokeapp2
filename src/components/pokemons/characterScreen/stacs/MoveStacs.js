import React, { useState } from 'react'
import PropTypes from "prop-types";

import { ExelColums } from '../../../commons/ExelColums';
import { DescriptionBox } from '../../../commons/DescriptionBox';

export const MoveStacs = ({
    move_damage_class: classMove,
    moves,
}) => {

    const [activeMove, setactiveMove] = useState({
        name: moves[0].name,
        url: moves[0].url
    })

    const handleClick = (elem) => {
        setactiveMove(s => ({
            ...s,
            name: elem.name,
            url: elem.url
        }))
    }
    return (
        <div className='typeStacs'>

            <div className='statsTitle'>
                <h4>Moves</h4>
            </div>

            <ExelColums
                key={classMove.name}
                item={'Damage Class'}
                values={[classMove]}
            />
            <div className='movesBox'>
                {
                    moves.map(elem =>
                    (<button
                        // className={(typeStacs.type === elem.type.name) ? 'active' : ''}
                        key={elem.name}
                        onClick={() => handleClick(elem)}
                    >{elem.name}</button>)
                    )
                }
            </div>

            <DescriptionBox data={activeMove} />
        </div>
    )
}

MoveStacs.propTypes = {
    move_damage_class: PropTypes.object.isRequired,
    moves: PropTypes.array.isRequired
}
