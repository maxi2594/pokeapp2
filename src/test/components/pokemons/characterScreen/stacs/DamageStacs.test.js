import { render } from "@testing-library/react"
import { shallow } from "enzyme"
import { DamageStacs } from "../../../../../components/pokemons/characterScreen/stacs/DamageStacs"
import { data } from "../../../../fakeData/fakeData"



describe('prubas en <DamageStacs />', () => {

    test('should render properly', () => {

        const wrapper = shallow(
            <DamageStacs damage={{ ...data.damage_relations }} />
        )

        expect(wrapper).toMatchSnapshot()
    })

    test('should have every damage titles and values', (done) => {

        const { getByText, container } = render(
            <DamageStacs damage={{ ...data.damage_relations }} />
        )

        const titles = Object.keys(data.damage_relations);

        titles.forEach((elem) => {
            expect(getByText(new RegExp(elem.toString(), 'i'))).toBeInTheDocument()
        })

        titles.forEach((elem) => {
            const spans = container.getElementsByClassName(elem);
            const dataValues = data.damage_relations[elem];

            expect(spans.length).toEqual(dataValues.length)
        })

        done()
    })


})
