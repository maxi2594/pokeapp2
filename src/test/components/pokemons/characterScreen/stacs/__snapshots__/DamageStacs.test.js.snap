// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prubas en <DamageStacs /> should render properly 1`] = `
<div
  className="typeStacs"
>
  <div
    className="statsTitle"
  >
    <h4>
      Damage Stacs
    </h4>
  </div>
  <ExelColums
    item="double_damage_from"
    key="double_damage_from"
    values={
      Array [
        Object {
          "name": "fighting",
          "url": "https://pokeapi.co/api/v2/type/2/",
        },
        Object {
          "name": "ground",
          "url": "https://pokeapi.co/api/v2/type/5/",
        },
        Object {
          "name": "steel",
          "url": "https://pokeapi.co/api/v2/type/9/",
        },
        Object {
          "name": "water",
          "url": "https://pokeapi.co/api/v2/type/11/",
        },
        Object {
          "name": "grass",
          "url": "https://pokeapi.co/api/v2/type/12/",
        },
      ]
    }
  />
  <ExelColums
    item="double_damage_to"
    key="double_damage_to"
    values={
      Array [
        Object {
          "name": "flying",
          "url": "https://pokeapi.co/api/v2/type/3/",
        },
        Object {
          "name": "bug",
          "url": "https://pokeapi.co/api/v2/type/7/",
        },
        Object {
          "name": "fire",
          "url": "https://pokeapi.co/api/v2/type/10/",
        },
      ]
    }
  />
</div>
`;
